cmd/kubectl-gadget/utils/trace.go

```diff
@@ -374,59 +377,101 @@ func SetTraceOperation(traceID string, operation string) error {
 	return err
 }
 
-// waitForTraceState loops over all trace whom ID is given as parameter
-// waiting until they are in the expected state.
-func waitForTraceState(traceID string, expectedState string) (*gadgetv1alpha1.TraceList, error) {
-	start := time.Now()
+// getTraceWatcher returns a watcher on trace(s) whom ID was given as parameter.
+// This watcher can then be used to wait on State.Output modification.
+func getTraceWatcher(traceID string) (watch.Interface, error) {
+	traceClient, err := getTraceClient()
+	if err != nil {
+		return watch.NewEmptyWatch(), err
+	}
 
-RetryLoop:
-	for {
-		successNodeCount := 0
-		timeout := time.Since(start) > traceTimeout
-		nodeErrors := make(map[string]string)
-		nodeWarnings := make(map[string]string)
+	var watchOptions = metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", GLOBAL_TRACE_ID, traceID),
+	}
 
-		traces, err := getTraceListFromID(traceID)
-		if err != nil {
-			return nil, err
+	watcher, err := traceClient.GadgetV1alpha1().Traces("gadget").Watch(context.TODO(), watchOptions)
+	if err != nil {
+		return watch.NewEmptyWatch(), err
+	}
+
+	return watcher, nil
+}
+
+// waitForTraceState wait for traces whom ID is given as parameter to be in the
+// expected state.
+func waitForTraceState(traceID string, expectedState string) (gadgetv1alpha1.TraceList, error) {
```

Comment:
I changed most of the methods to return `*gadgetv1alpha1.TraceList` in https://github.com/kinvolk/inspektor-gadget/pull/416, would you mind doing the same for this one?